// edition: 2021
#![deny(warnings)]
#![doc(no_auto_cfg)]
/*#![doc(no_auto_cfg(1))] //~ ERROR
#![doc(no_auto_cfg = 1)] //~ ERROR
#![doc(auto_cfg(1))] //~ ERROR
#![doc(auto_cfg = 1)] //~ ERROR

#[doc(auto_cfg)] //~ ERROR
//~^ WARN
#[doc(no_auto_cfg)] //~ ERROR
//~^ WARN
pub struct Bar;*/
